web/src/API.re

```diff
@@ -0,0 +1,42 @@
+open Relude.Globals;
+
+module Error = {
+  type t = ReludeFetch.Error.t(string);
+  let show = error => ReludeFetch.Error.show(a => a, error);
+  module Type = {
+    type nonrec t = t;
+  };
+};
+
+module Endpoint = {
+  type t =
+    | URLs;
+
+  let unwrap =
+    fun
+    | URLs => "/api/url/list";
+};
+
+module URLs = {
+  [@decco]
+  type url = {
+    id: int,
+    url: string,
+    slug: string,
+    destination: string,
+    ts: string,
+  };
+
+  [@decco]
+  type t = {data: option(list(url))};
+
+  let decode: Js.Json.t => Belt.Result.t(t, string) =
+    json => json->t_decode |> Relude.Result.mapError(_ => "Decode failed");
```

Comment:
can you see `t_decode` (we never define or write `t_decode`), but this generated by `[@decco]`